src/dataset-builder/DomainCriteriaSelector.ts

```diff
@@ -59,19 +64,41 @@ export const saveSelected =
 
 export const DomainCriteriaSelector = (props: DomainCriteriaSelectorProps) => {
   const [rootConcepts, loadRootConcepts] = useLoadedData<GetConceptsResponse>();
+  const [hierarchyConcepts, loadHierarchyConcepts] = useLoadedData<GetConceptsHierarchyMapResponse>();
   const { state, onStateChange, datasetId, getNextCriteriaIndex } = props;
+  const [hierarchy, setHierarchy] = useState(new Map<number, Concept[]>());
+  const [isHierarchyLoaded, setIsHierarchyLoaded] = useState(false);
   useOnMount(() => {
-    void loadRootConcepts(() => DataRepo().dataset(datasetId).getConcepts(state.domainOption.root));
+    const selectedConcept = state.selectedConcept;
+    if (selectedConcept) {
+      void loadHierarchyConcepts(() => DataRepo().dataset(datasetId).getConceptsHierarchy(selectedConcept));
+    } else {
+      // get me the children of this concept id
+      void loadRootConcepts(() => DataRepo().dataset(datasetId).getConcepts(state.domainOption.root));
```

Comment:
should this be rendering the search state if no concept is selected/opened? Would it be something like this?
void loadRootConcepts(() => DataRepo().dataset(datasetId).searchConcepts(state.domainOption.root, ''));